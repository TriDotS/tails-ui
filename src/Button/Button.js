import React from 'react'
import styled from 'styled-components'

const Button = styled.button`
  display: inline-block;
  padding: 10px 10px;
  font-size: 1em;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
  outline: none;
  color: ${props => props.fontColor || 'black'};
  border-radius: ${props => (props.rounded ? '3px' : '0px')};
  background: ${props =>
    props.hollow ? 'transparent' : props.backgroundColor || 'palevioletred'};
  border: ${props =>
    props.hollow ? `2px solid ${props.borderColor}` : 'none'};
  box-shadow: ${props => (props.shadowed ? '0 3px dimgrey' : null)};
  width: ${props => (props.fullWidth ? '100%' : null)};

  &:active {
    box-shadow: 0;
    transform: translateY(2px);
    filter: brightness(85%);
  }
`

export default ({
  children,
  onClick,
  type,
  fullWidth,
  backgroundColor,
  fontColor,
  shadowed,
  rounded,
  hollow,
  borderColor,
}) =>
  <Button
    type={type}
    fullWidth={fullWidth}
    backgroundColor={backgroundColor}
    fontColor={fontColor}
    onClick={onClick}
    shadowed={shadowed}
    rounded={rounded}
    hollow={hollow}
    borderColor={borderColor}
  >
    {children}
  </Button>
