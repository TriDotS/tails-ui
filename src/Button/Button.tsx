import * as React from 'react'
import styled from 'styled-components'

const StyledButton = styled.button`
  background: blue;
  color: white;
  border: none;
`

export const Button: React.SFC<any> = () => <StyledButton>Buttton</StyledButton>
