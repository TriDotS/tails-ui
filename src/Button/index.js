import React from 'react'
import styled from 'styled-components'

const ButtonContainer = styled.button`
  background: olive;
  font-weight: bolder;
  padding: 4px;
  border: none;
  &:focus {
    outline: none;
  }
`

const Button = () => (
  <ButtonContainer>
    Button
  </ButtonContainer>
)

export default Button