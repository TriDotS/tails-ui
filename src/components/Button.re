let component = ReasonReact.statelessComponent("Button");

let make =
    (
      ~fontColor,
      ~loading,
      ~width,
      ~backgroundColor,
      ~borderColor,
      ~borderRadius,
      ~children
    ) => {
  ...component,
  render: _self =>
    <button
      className={j|$(borderRadius) border-2 border-$(borderColor) w-$(width) cursor-pointer bg-$(backgroundColor) hover:bg-$(backgroundColor)-dark text-$(fontColor) font-sans font-bold py-2 px-4 shadow inline-flex items-center justify-center|j}>
      (
        loading === true ?
          ReasonReact.stringToElement("loading...") :
          ReasonReact.arrayToElement(children)
      )
    </button>
};

let default =
  ReasonReact.wrapReasonForJs(~component, jsProps =>
    make(
      ~fontColor=jsProps##fontColor,
      ~loading=jsProps##loading,
      ~width=jsProps##width,
      ~backgroundColor=jsProps##backgroundColor,
      ~borderColor=jsProps##borderColor,
      ~borderRadius=jsProps##borderRadius,
      ~children=jsProps##children
    )
  );
