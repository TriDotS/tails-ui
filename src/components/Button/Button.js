import PropTypes from 'prop-types'
import React from 'react'

const Button = ({
  children,
  color,
  loading,
  outline,
  fullWidth,
  rounded,
  ...props
}) => (
  <button
    className={
      outline
        ? `${fullWidth &&
            'w-full'} cursor-pointer bg-transparent hover:bg-${color} text-${color}-dark font-semibold hover:text-white py-2 px-4 border border-${color} hover:border-transparent ${
            rounded ? 'rounded-full' : 'rounded'
          } inline-flex items-center justify-center`
        : `${fullWidth &&
            'w-full'} cursor-pointer bg-${color} hover:bg-${color}-dark text-white font-sans font-bold py-2 px-4 shadow ${
            rounded ? 'rounded-full' : 'rounded'
          } inline-flex items-center justify-center`
    }
    {...props}
  >
    {loading ? 'loading...' : children}
  </button>
)

Button.defaultProps = {
  color: 'teal'
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  color: PropTypes.string,
  loading: PropTypes.bool,
  outline: PropTypes.bool,
  fullWidth: PropTypes.bool,
  rounded: PropTypes.bool
}

export default Button
