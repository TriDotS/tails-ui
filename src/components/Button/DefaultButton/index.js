import PropTypes from 'prop-types'
import React from 'react'

const DefaultButton = ({ children }) => (
  <button className="bg-grey hover:bg-grey-dark text-white font-bold py-2 px-4 rounded">
    {children}
  </button>
)

DefaultButton.propTypes = {
  children: PropTypes.node
}

export default DefaultButton
