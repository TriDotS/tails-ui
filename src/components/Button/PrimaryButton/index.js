import PropTypes from 'prop-types'
import React from 'react'

const PrimaryButton = ({ children }) => (
  <button className="bg-blue hover:bg-blue-dark text-white font-bold py-2 px-4 rounded">
    {children}
  </button>
)

PrimaryButton.propTypes = {
  children: PropTypes.node
}

export default PrimaryButton
