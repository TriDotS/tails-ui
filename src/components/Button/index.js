import DefaultButton from './DefaultButton'
import PrimaryButton from './PrimaryButton'
import PropTypes from 'prop-types'
import React from 'react'
import SecondaryButton from './SecondaryButton'

const Button = ({
  children,
  primary,
  secondary,
  success,
  info,
  danger,
  loading
}) => {
  if (primary) {
    return <PrimaryButton>{loading ? 'loading...' : children}</PrimaryButton>
  }
  if (secondary) {
    return (
      <SecondaryButton>{loading ? 'loading...' : children}</SecondaryButton>
    )
  }
  return <DefaultButton>{loading ? 'loading...' : children}</DefaultButton>
}

Button.propTypes = {
  children: PropTypes.node,
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  success: PropTypes.bool,
  info: PropTypes.bool,
  danger: PropTypes.bool,
  loading: PropTypes.bool
}

export default Button
