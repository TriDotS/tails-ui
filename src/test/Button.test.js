import Button from '../components/Button'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Button>Button</Button>, div)
})

test('button snapshot', () => {
  const tree = renderer.create(<Button>Button</Button>).toJSON()
  expect(tree).toMatchSnapshot()
})
