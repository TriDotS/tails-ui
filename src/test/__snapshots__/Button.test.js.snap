// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button snapshot 1`] = `
<div
  className="undefined cursor-pointer bg-teal hover:bg-teal-dark text-white font-sans font-bold py-2 px-4 shadow rounded inline-flex items-center justify-center"
>
  Button
</div>
`;
